Library/Homebrew/cmd/pyenv-sync.rb

```diff
@@ -0,0 +1,82 @@
+# typed: true
+# frozen_string_literal: true
+
+require "cli/parser"
+require "formula"
+
+module Homebrew
+  module_function
+
+  def pyenv_sync_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `pyenv-sync`
+
+        Create symlinks for Homebrew's installed Python versions in ~/.pyenv/versions.
+
+        Note that older patch version symlinks will be created and linked to the minor
+        version so e.g. Python 3.11.0 will also be symlinked to 3.11.3.
+      EOS
+
+      named_args :none
+    end
+  end
+
+  def pyenv_sync
+    dot_pyenv = Pathname(Dir.home)/".pyenv"
+
+    # Don't run multiple times at once.
+    pyenv_sync_running = dot_pyenv/".pyenv_sync_running"
+    return if pyenv_sync_running.exist?
+
+    @pyenv_versions = dot_pyenv/"versions"
+    @pyenv_versions.mkpath
+    FileUtils.touch pyenv_sync_running
+
+    pyenv_sync_args.parse
+
+    HOMEBREW_CELLAR.glob("python{,@*}")
+                   .flat_map(&:children)
+                   .sort_by(&method(:gem_like_version))
```

Comment:
Since we're simply iterating over each of these version paths, is there a specific reason to sort these by version?